### PR TITLE
Server uses single config instance internally.

### PR DESCRIPTION
**Pull request by jakewins**
This deprecates all public exposure of config APIs that are not Neo4j Config,
and modifies the server to use a single Config instance internally.

Config is loaded in the same way as before, from neo4j.properties and
neo4j-server.properties, but now neither file is special - any config can
conceivably be specified in each.

This is yet another step towards consolidating config, and a helpful step for
browser to access config specified in neo4j-server.properties, since those
settings will now be available via jmx.

### CODE REVIEW

File: community/server/src/main/java/org/neo4j/server/configuration/ServerSettings.java [hunk 1]
```diff
@@ -75,7 +75,7 @@
 
     @Description( "Number of Neo4j worker threads." )
     Setting<Integer> webserver_max_threads = setting( "org.neo4j.server.webserver.maxthreads",
-            INTEGER, NO_DEFAULT, min( 1 ) );
+            INTEGER, ""+Math.min(Runtime.getRuntime().availableProcessors(),500), min( 1 ) );
```
Comment (chrisvest) [inline community/server/src/main/java/org/neo4j/server/configuration/ServerSettings.java:82]: Was this defined elsewhere?

File: community/server/src/test/java/org/neo4j/server/modules/ThirdPartyJAXRSModuleTest.java [hunk 2]
```diff
@@ -54,18 +54,20 @@ public void shouldReportThirdPartyPackagesAtSpecifiedMount() throws Exception
...
         when( config.get( ServerSettings.third_party_packages ) ).thenReturn( jaxRsPackages );
-        when( neoServer.getConfigurationBuilder().configuration() ).thenReturn( config );
+//        when( neoServer.getConfigurationBuilder().configuration() ).thenReturn( config );
         
         ThirdPartyJAXRSModule module = new ThirdPartyJAXRSModule(webServer, config, NullLogProvider.getInstance(), neoServer );
         module.start();
 
         verify( webServer ).addJAXRSPackages( any( List.class ), anyString(), anyCollection() );
+
+        throw new RuntimeException( "Ha" );
```
Comment (chrisvest) [inline community/server/src/test/java/org/neo4j/server/modules/ThirdPartyJAXRSModuleTest.java]: What's going on here?

File: community/server/src/test/java/org/neo4j/server/webadmin/rest/VersionAndEditionServiceTest.java [hunk 3]
```diff
@@ -19,9 +19,11 @@
  */
 package org.neo4j.server.webadmin.rest;
 
-import javax.ws.rs.core.Response;
-
 import org.junit.Test;
+import sun.reflect.generics.reflectiveObjects.NotImplementedException;
```
Comment (MishaDemianenko) [inline community/server/src/test/java/org/neo4j/server/webadmin/rest/VersionAndEditionServiceTest.java]: Can we switch to `java.lang.UnsupportedOperationException` instead?

Comment (jakewins) [inline community/server/src/test/java/org/neo4j/server/webadmin/rest/VersionAndEditionServiceTest.java]: +1

File: community/server/src/test/java/org/neo4j/server/modules/ThirdPartyJAXRSModuleTest.java (hunk 2 above)
Comment (jakewins) [inline community/server/src/test/java/org/neo4j/server/modules/ThirdPartyJAXRSModuleTest.java]: "Ha", apparently. I left that there to remind myself to fix that test before I made this PR, evidently that did not work.. :)

File: community/server/src/main/java/org/neo4j/server/configuration/ServerSettings.java (hunk 1 above)
Comment (jakewins) [inline community/server/src/main/java/org/neo4j/server/configuration/ServerSettings.java:82]: Yes, this used to be in AbstractNeoServer as a method of its own.

**Comment by jakewins** [comment]
Should not be merged before https://github.com/neo4j/neo4j/pull/4839

**Comment by jakewins** [comment]
/cc @benbc @cleishm note that this PR also proposes to add two (unofficial) command line flags to the server bootstrappers, `-C` and `-c`. `-C` specifies a path to a config file, `-c` allows overriding config using the 'metadata' option approach we use elsewhere:

```
java -jar neo4j-server.jar -C /etc/neo4j.conf -c xx.ndp.enabled=true
```

This isn't exposed in any public surface, but can be (initially at least) used by the shell scripts to a set config file location. It's part of this PR because I wanted to test setting config location in tests without overriding global system properties. 

My suggestion in the medium term though is that we will remove command line parsing from the shell scripts, testing and handling that instead in a single place. Obviously the exact flags and potential commands needs a larger discussion (as does the overall approach).

**Comment by cleishm** [comment]
@jakewins That's cool. Although we did discuss that we'd only be able to use that approach if we introduce a way for java to re-launch itself after having read the various JVM properties. Until then, the shell script needs to read the config also, and thus needs to know where it is. I'm still 50/50 on whether doing that in re-launching in Java is a good idea (vs shell), given launching process is exactly what shell script is designed for! But we should spike something out together sometime.

**Comment by davidegrohmann** [comment]
@jakewins this needs a rebase

**Comment by jakewins** [comment]
@cleishm agreed, I'm not sure this is the right approach - but if it works to take what you did in the shell and do it directly in Java, it'd be nice wrt cross-platform stuff and so on. 

In either case, I wanted to give you a heads up that there's a nibble of that in here, but that I'm happy to roll it back if we don't like it. Since it's non-public and it helps with testing, I think it's ok for now and we can see which direction we take this further down the line?

**Comment by cleishm** [comment]
+1

**Pull request merged**